app/models/metric/purging.rb

```diff
@@ -50,56 +48,84 @@ def self.purge_timer(ts, interval)
     end
   end
 
-  def self.scope_for_interval(interval)
-    interval == 'realtime' ? Metric : MetricRollup.where(:capture_interval_name => interval)
+  def self.purge_window_size
+    VMDB::Config.new("vmdb").config.fetch_path(:performance, :history, :purge_window_size) || 1000
+  end
+
+  def self.purge_conditions(older_than, interval)
+    scope = interval == 'realtime' ? Metric : MetricRollup.where(:capture_interval_name => interval)
+    scope.where(scope.arel_table[:timestamp].lteq(older_than))
   end
 
   def self.purge_count(older_than, interval)
-    scope_for_interval(interval).where('timestamp <= ?', older_than).count
+    purge_conditions(older_than, interval).count
   end
 
-  def self.purge(older_than, interval, window = nil, limit = nil)
-    _log.info("Purging #{limit.nil? ? "all" : limit} #{interval} metrics older than [#{older_than}]...")
+  def self.purge_associated_records(metric_type, ids)
+    # Since VimPerformanceTagValues are 6 * number of tags per performance
+    # record, we need to batch in smaller trips.
+    count_tag_values = 0
+    _log.info("Purging associated tag values.")
+    ids.each_slice(50) do |vp_ids|
```

Comment:
I looked at the code and it does shorten it a bit, but it is a little different. it sends all the ids to the server every time rather than only sending 50 at a time.

Is this what you had in mind?

```ruby
  def self.purge_associated_records(metric_type, ids)
    # Since VimPerformanceTagValues are 6 * number of tags per performance
    # record, we need to batch in smaller trips.
    _log.info("Purging associated tag values.")
    VimPerformanceTagValue.where(:metric_id => vp_ids, :metric_type => metric_type)
      .delete_in_batches(50).tap do |count|
        _log.info("Purged #{count_tag_values} associated tag values.")
      end
  end
```